File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums,int idx=0) {
        int n = nums.size() ;
        if(idx == n-1) return true ;
        if(nums[idx] == 0 ) return false ;

        for(int i=1;i<=nums[idx];i++){
            if(canJump(nums,idx+i)) return true ;
        }

        return false ;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        
        self.st = set()
        n = len(nums)
        nums.sort()
        def dp(i, curr):
            if i == n:
                self.st.add(tuple(curr))
                return
            curr.append(nums[i])
            dp(i+1, curr)
            curr.pop()
            dp(i+1, curr)
            return

        dp(0,[])

        return list(self.st)
